nemo-physical/src/dictionary/datavalue_dictionary.rs

```diff
@@ -65,6 +67,22 @@ pub trait DvDict: Debug + ByteSized {
     /// with [KNOWN_ID_MARK].
     fn add_datavalue(&mut self, dv: AnyDataValue) -> AddResult;
 
+    /// Returns a function pointer to a function that takes a (parent) [MetaDvDictionary] and a dictionary id
+    /// that identifies this dictionary rather than `&self`. Mutable access to a parent dictionary is required
+    /// to resolve nested terms recursively, but making this a method (that takes `&self`) would offend the Rust
+    /// borrow checker even if self is never accessed directly within the method. Returning a function pointer
+    /// replaces the native dynamic dispatch without requiring `&self` to be present.
+    ///
+    /// The default implementation of this method returns a pointer to a function that
+    /// simply calls [DvDict::add_datavalue] after retrieving the identified dictionary from the parent.
+    /// There is no major performance penalty in this double indirection since [MetaDvDictionary] can cache the
+    /// function pointer, and would have to retrieve the sub-dictionary anyway to call it.
+    fn add_datavalue_with_parent_fn(
+        &self,
+    ) -> fn(&mut MetaDvDictionary, dict_id: usize, dv: AnyDataValue) -> AddResult {
```

Comment:
this is a stylistic choice, but you could also implement those function pointers as closures (since they can be converted into fn pointers automatically, as long as they don't capture anything).

Reply:
What would be the benefit of doing this?

Reply:
Ah, maybe I get it. You want to "inline" the functions in this way. The signature of the method would stay but the functions move "up" in the source file? Maybe one could try that, but the more complex cases will use auxiliary (private) functions that would have to be outside of the trait implementation, in this case leading to larger distance between related code n the source file.

Reply:
Ok, so I did this now for the short default implementations in this file. I am not convinced that the same approach would make the longer code in tuple_dv_dict.rs more readable though. One of the callbacks there also has own documentation that describes when it will panic; would probably not fit in front of a closure.